common/lib/container-definitions/src/loader.ts

```diff
@@ -475,7 +476,22 @@ declare module "@fluidframework/core-interfaces" {
     }
 }
 
+/**
+ * @deprecated 0.48, This API will be removed in 0.50
+ * No replacement since it is not expected anyone will depend on this outside container-loader
+ * See https://github.com/microsoft/FluidFramework/issues/9711 for context
+ */
 export interface IPendingLocalState {
     url: string;
     pendingRuntimeState: unknown;
 }
+
+/**
+ * This is used when we rehydrate a container from the snapshot. Here we put the blob contents
+ * in separate property: blobContents. This is used as the ContainerContext's base snapshot
+ * when attaching.
+ */
+export interface ISnapshotTreeWithBlobContents extends ISnapshotTree {
```

Comment:
I see exactly same interface in packages\loader\container-loader\src\utils.ts. Can we dedup that (eventually)?

Reply:
Yes, this needs to go in first though